### PR TITLE
test against production assets (#133)

### PR DESCRIPTION
**Pull request by bollwyvl**
## Demo

[Binder](https://mybinder.org/v2/gh/bollwyvl/jupyterlab-lsp/test-minified-assets?urlpath=lab)

## References

- informed by #133

## Code changes

This uses `--dev-build=False --minimize=True` for acceptance test activities. These flags were not available on some older versions of lab, so an `npm` flake might cause issues for the older excursions. It _will_ take more time. However, this is how users will actually have to deal with the extension(s) as installed, so _not_ testing it is somewhat reckless.

## User-facing changes

None

## Backwards-incompatible changes

None

## Chores

- [x] linted
- [ ] tested
- [x] documented
- [ ] changelog entry

### CODE REVIEW

**Comment by bollwyvl** [comment]
Linted, and in the process also clobbered some new warnings from `pytest` and `pytest_azurepipelines`).

**Comment by bollwyvl** [comment]
Looks like it takes about 30 seconds longer to do the build on windows. Also didn't actually push my lint, which spiraled into more linting. And squashing the warnings created new warnings. Anyhow, if it comes up clean (with or without warnings, I guess) this is good to go, as I have to get going.

**Comment by bollwyvl** [comment]
Given the number of things that conflicted between `black`, `flake8`, etc. I've gone ahead and formalized the linting i have been doing when i was doing the initial work on the python side.

File: atest/01_Editor.robot [hunk 1]
```diff
@@ -21,17 +21,15 @@ Bash
 
 CSS
     ${def} =    Set Variable    xpath:(//span[contains(@class, 'cm-variable-2')][contains(text(), '--some-var')])[last()]
-    Editor Shows Features for Language    CSS    example.css    Diagnostics=Do not use empty rulesets    Jump to Definition=${def}
-    ...    Rename=${def}
+    Editor Shows Features for Language    CSS    example.css    Diagnostics=Do not use empty rulesets    Jump to Definition=${def}    Rename=${def}
```
Comment (krassowski) [inline atest/01_Editor.robot]: IMO The best thing would be to have the keyword arguments in separate lines, as right now it is difficult to read and see which ones are there. Like:
```
Editor Shows Features for Language    CSS    example.css
...    Diagnostics=Do not use empty rulesets
...    Jump to Definition=${def}
...    Rename=${def}
```
I do not suggest that you take any action (I guess this ain't as intelligent linter as black), just complaining about life.

**Comment by bollwyvl** [comment]
Ooh, yeah, that's gross.

Maybe we move the whole shooting match to a python file. There comes a time
when robot is too cute.

I think there was a pr upstream for that... But my associate and on the
robots-from-jupyter stuff have considered doing a language server based on
pylgs, and would definitely like to have formatting that we'd actually want
to use, in addition to rflint and the dry run tools.

**Comment by krassowski** [comment]
Thank you, looks great to me!

Obviously, my predilection for single quotes will remain unsatisfied ;)

File: environment-lint.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+channels:
+  - conda-forge
+  - defaults
+
+dependencies:
+  - black
+  - isort
+  - mypy
+  - robotframework-lint
```
Comment (krassowski) [inline environment-lint.yml]: Also needed in `requirements-dev.txt`, right?

**Comment by krassowski** [comment]
I wonder if I broke master, or if it's just something with the tests for merges:

Screenshot from 2020-01-07 16-17-33

:confused:

**Pull request merged**